intro.md

```diff
@@ -1,9 +1,21 @@
 # Introduction
 
 ## Design patterns
-
-What are design patterns? What are idioms? Anti-patterns.
+When developing programs, we have to solve many problems. A program can be viewed as a solution to a problem. It can also be viewed as a collection of solutions to many different problems. All of these solutions work together to solve a bigger problem.
 
 ## Design patterns in Rust
 
-Why Rust is a bit special - functional elements, type system - borrow checker
+There are many problems that share the same form. Due to the fact that Rust is not object-oriented design patterns vary with respect to other object-oriented programming languages. While the details are different, since they have the same form they can be solved using the same fundamental methods. 
+
+[Design patterns](patterns/README.md) are methods to solve common problems when writing software.
+
+[Anti-patterns](anti_patterns/README.md) are methods to solve these same common problems. 
```

Comment:
Feels like the same wordings, maybe we should let readers know what are the differences between them? "Anti-patterns" is a tough english word, we should have more descriptions.